File: AVCaptureDemo/ViewController.m
//
//  ViewController.m
//  AVCaptureDemo
//
//  Created by yunfu on 2019/2/28.
//  Copyright © 2019 yunfu. All rights reserved.
//

#import "ViewController.h"
#import "HLCapture/HLCapture.h"
@interface ViewController ()<HLTakePhotoDelegate>
@property (weak, nonatomic) IBOutlet UIImageView *imageView;

@end

@implementation ViewController
{
    HLCapture *capture;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    capture = [[HLCapture alloc] initWithType:HLCaptureTypeVideo];
    [capture prepareWithSize:self.view.frame.size];
    capture.takePhotoDelegate = self;
    [self.view addSubview:capture.videoView];
    
    [self.view sendSubviewToBack:capture.videoView];
    
}

- (void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    [capture startCapture];
}

- (IBAction)takeClick:(id)sender {
    [capture takePhoto];
}


- (void)takePhoto:(NSData *)imageData{
    self.imageView.image = [[UIImage alloc] initWithData:imageData];
}
- (IBAction)flashClick:(id)sender {
    [capture openAndCloseTorch];
}
- (IBAction)switchCamera:(id)sender {
    [capture switchCameras];
}


@end

File: AVCaptureDemo_Swift/ViewController.swift
//
//  ViewController.swift
//  AVCaptureDemo_Swift
//
//  Created by yunfu on 2019/3/1.
//  Copyright © 2019 yunfu. All rights reserved.
//

import UIKit
import AVFoundation

class ViewController: UIViewController,HLTakePhotoDelegate {

    @IBOutlet weak var imageView: UIImageView!
    let captureManager = HLCaptureManager(.video)
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let preView = HLCaptureView(frame:CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: self.view.frame.size.height))
        self.view.addSubview(preView)
        captureManager.prepare()
        preView.session = captureManager.session
        captureManager.takePhotoDelegate = self
        self.view.sendSubviewToBack(preView)
        captureManager.startRunning()
    }
    
    func takePhoto(photoData: Data) {
        self.imageView.image = UIImage(data: photoData)
    }


    
    @IBAction func takePhoto(_ sender: Any) {
        captureManager.take()
    }
    
    @IBAction func torchClick(_ sender: Any) {
        captureManager.openAndCloseTorch()
    }
    
    @IBAction func switchClick(_ sender: Any) {
        captureManager.switchCameras()
    }
}

